### ISSUE TITLE
Random failures on CI from the bhttp library, used by the broken link checker

### ISSUE THREAD

**Issue by emmenko** [commercetools/commercetools-docs-kit#385]
We keep getting some errors on CI from time to time regarding the broken link checker

```
- Link Checking public/docs-smoke-test/ {http://localhost:40531/docs-smoke-test/} -
/home/runner/work/commercetools-docs-kit/commercetools-docs-kit/node_modules/bhttp/lib/bhttp.js:159
  err.response = response;
               ^

TypeError: Cannot assign to read only property 'response' of object '[object Object]'
    at addErrorData (/home/runner/work/commercetools-docs-kit/commercetools-docs-kit/node_modules/bhttp/lib/bhttp.js:159:16)
    at ClientRequest.<anonymous> (/home/runner/work/commercetools-docs-kit/commercetools-docs-kit/node_modules/bhttp/lib/bhttp.js:567:25)
    at ClientRequest.emit (events.js:310:20)
    at TLSSocket.socketErrorListener (_http_client.js:426:9)
    at TLSSocket.emit (events.js:310:20)
    at emitErrorNT (internal/streams/destroy.js:92:8)
    at emitErrorAndCloseNT (internal/streams/destroy.js:60:3)
    at processTicksAndRejections (internal/process/task_queues.js:84:21)
```

Restarting the build usually helps, which hints that the problem appears randomly.

Not sure if we can do much about it, but we can at least try to look into it.

**Comment by emmenko** [comment]
We could also look into another "broken link checker" library that is more maintained.

**Comment by nkuehn** [comment]
There's another very similar lib & CLI called "linkinator", which is more active but also a one-man-show. 

The author of broken-link-checker  has started a rework last summer as an alpha but didn't complete it as it seems. 

That's always the issue with one man shows.  one or two years the topic is relevant to them and then something changes (job, kids, ..) and the project is left alone.

**Comment by nkuehn** [comment]
It's gotten worse with the full content of the site, It doesn't finish most of the time.  Sad, but I think we have to tackle this faster than hoped for

**Comment by nkuehn** [comment]
Update @davifantasia :  I have tried the "linkinator" tool as a CLI on the complete migrated docs and it seems to be stable and fast.  

The default output is not perfect but good enough to go with so I suggest to not even build a custom wrapper again and use it OOTB, rather submitting PRs there.  

Issues:
 * we can't warn on unsecure links and 
 * the output contains the temporary local host and port which will confuse some users.  
 * I have to create an index.html file with links in the root directory so that it finds the sites in the folders. but again, that's not a real blocker. 

Overall I will try it out now in the private content repo and if we're happy we can decide to remove the broken-link-checker package from the kit repo altogether and deprecate the package on NPM. 

PS: why not from the beginning?  The linkinator project was just started in Feb 2019 so I did not find it, either because it wasn't there or not famous enough yet.
